app/src/main/java/org/literacyapp/authentication/DetectionHelper.java

```diff
@@ -17,13 +17,17 @@
     private static final Scalar RED_COLOR = new Scalar(255, 0, 0, 255);
 
     public static boolean isFaceInsideFrame(AnimalOverlay animalOverlay, Mat img, Rect face){
-        Point frameTopLeft = new Point(animalOverlay.getFrameStartX(), animalOverlay.getFrameStartY());
-        Point frameBottomRight = new Point(animalOverlay.getFrameEndX(), animalOverlay.getFrameEndY());
-        Rect frame = new Rect(frameTopLeft, frameBottomRight);
-        if (face.tl().x >= frame.tl().x && face.tl().y >= frame.tl().y && face.br().x <= frame.br().x && face.br().y <= frame.br().y){
-            return true;
+        if (animalOverlay != null){
+            Point frameTopLeft = new Point(animalOverlay.getFrameStartX(), animalOverlay.getFrameStartY());
+            Point frameBottomRight = new Point(animalOverlay.getFrameEndX(), animalOverlay.getFrameEndY());
+            Rect frame = new Rect(frameTopLeft, frameBottomRight);
+            if (face.tl().x >= frame.tl().x && face.tl().y >= frame.tl().y && face.br().x <= frame.br().x && face.br().y <= frame.br().y){
```

Comment:
Add parenthesises to improve readability: `((face.tl().x >= frame.tl().x) && ...`